src/jit/compiler.hpp

```diff
@@ -2768,7 +2769,28 @@ inline bool Compiler::fgIsThrowHlpBlk(BasicBlock* block)
         return false;
     }
 
-    GenTreePtr call = block->bbTreeList->gtStmt.gtStmtExpr;
+    GenTree* call = nullptr;
+    if (block->IsLIR())
+    {
+        LIR::Range& blockRange = LIR::AsRange(block);
+        call                   = blockRange.LastNode();
```

Comment:
Yes, I just realized that it already exists. It would seem better to use that, and then add a debug check that, if there is a call, it is the only call in the block.

---
In reply to: [75210401](https://github.com/dotnet/coreclr/pull/6689#discussion_r75210401) [](ancestors = 75210401)